### PR TITLE
[#18] Double to float migration

### PR DESCRIPTION
**Pull request by ilia-glushchenko**

### CODE REVIEW

File: Epona/include/Epona/Analysis.hpp [hunk 1]
```diff
@@ -308,7 +308,7 @@ void CalculateCrossProductForeach(
  *
  * @tparam VectorType GLM vector type
  * @tparam InIterator forward iterator from the container of VectorType objects
- * @tparam OutIterator forward iterator from the container of double or float
+ * @tparam OutIterator forward iterator from the container of float or float
```
Comment (MrDmitry) [inline Epona/include/Epona/Analysis.hpp]:
```suggestion
 * @tparam OutIterator forward iterator for the container of float
```

File: Epona/include/Epona/Analysis.hpp [hunk 2]
```diff
@@ -308,7 +308,7 @@ void CalculateCrossProductForeach(
  *
  * @tparam VectorType GLM vector type
  * @tparam InIterator forward iterator from the container of VectorType objects
```
Comment (MrDmitry) [inline Epona/include/Epona/Analysis.hpp]:
```suggestion
 * @tparam InIterator forward iterator for the container of VectorType objects
```

File: Epona/include/Epona/FloatingPoint.hpp [hunk 3]
```diff
@@ -67,7 +67,7 @@ inline bool IsNotEqual(double lhs, double rhs)
  *
  *  @return @c true if lhs is less than rhs, @c false otherwise
  */
-inline bool IsLess(double lhs, double rhs)
+inline bool IsLess(float lhs, float rhs)
 {
     return !IsEqual(lhs, rhs) && (lhs - rhs) < 0.0;
```
Comment (MrDmitry) [inline Epona/include/Epona/FloatingPoint.hpp]:
```suggestion
    return !IsEqual(lhs, rhs) && (lhs - rhs) < 0.0f;
```

File: Epona/include/Epona/FloatingPoint.hpp [hunk 4]
```diff
@@ -81,7 +81,7 @@ inline bool IsLess(double lhs, double rhs)
  *
  *  @return @c true if lhs is greater than rhs, @c false otherwise
  */
-inline bool IsGreater(double lhs, double rhs)
+inline bool IsGreater(float lhs, float rhs)
 {
     return !IsEqual(lhs, rhs) && (lhs - rhs) > 0.0;
```
Comment (MrDmitry) [inline Epona/include/Epona/FloatingPoint.hpp]:
```suggestion
    return !IsEqual(lhs, rhs) && (lhs - rhs) > 0.0f;
```

File: Epona/include/Epona/JacobiEigenvalue.hpp [hunk 5]
```diff
@@ -23,37 +23,37 @@ class JacobiEigenvalue
      * @param[in] maxIterations
      */
     explicit JacobiEigenvalue(
-        glm::dmat3 const& symmetricMatrix,
-        double coverageThreshold = 1.0e-4,
+        glm::mat3 const& symmetricMatrix,
+        float coverageThreshold = 1.0e-4f,
```
Comment (MrDmitry) [inline Epona/include/Epona/JacobiEigenvalue.hpp]: Should this relate to `fp::g_floatingPointThreshold` or is it completely separate independent value?

**Review by MrDmitry — CHANGES_REQUESTED** [review]

File: Epona/include/Epona/JacobiEigenvalue.hpp (hunk 5 above)
Comment (ilia-glushchenko) [reply Epona/include/Epona/JacobiEigenvalue.hpp]: As I vaguely remember it probably should.

**Review by MrDmitry — APPROVED** [review]

**Pull request merged**